packages/frontend/src/lib/RecipeDetails.svelte

```diff
@@ -0,0 +1,137 @@
+<script lang="ts">
+import { faPlay, faRefresh } from '@fortawesome/free-solid-svg-icons';
+import { faGithub } from '@fortawesome/free-brands-svg-icons';
+import { getDisplayName } from '/@/utils/versionControlUtils';
+import type { Recipe } from '@shared/src/models/IRecipe';
+import { recipes } from '/@/stores/recipe';
+import Button from '/@/lib/button/Button.svelte';
+import TasksProgress from '/@/lib/progress/TasksProgress.svelte';
+import Fa from 'svelte-fa';
+import { studioClient } from '/@/utils/client';
+import { catalog } from '/@/stores/catalog';
+
+export let recipeId: string;
```

Comment:
Not required as id is already present in the Recipe type

Reply:
In the recipe interface, the property id is optional, therefore we either need to make a huge {#try}{/if} or improve the Recipe Interface to make he id  a mandatory property

Reply:
I fixed it by removing the recipe property @jeffmaury